go-controller/pkg/clustermanager/network_cluster_controller.go

```diff
@@ -202,6 +224,90 @@ func (ncc *networkClusterController) init() error {
 	return nil
 }
 
+// updateNetworkStatus allows to report an status for networkClusterController's network via a UDN status condition
+// of type "NetworkAllocationSucceeded", if the network was created by UDN.
+// When at least one node reports an error, condition will be set to false and an event with node-specific error will be
+// generated.
+// Call this function after every node event handling, set handlerErr to nil to report no error.
+func (ncc *networkClusterController) updateNetworkStatus(nodeName string, handlerErr error) error {
+	if ncc.statusReporter == nil {
+		return nil
+	}
+	errorMsg := ""
+	if handlerErr != nil {
+		errorMsg = handlerErr.Error()
+	}
+
+	ncc.nodeErrorsLock.Lock()
+	defer ncc.nodeErrorsLock.Unlock()
+	if ncc.nodeErrors[nodeName] == errorMsg {
+		// error message didn't change for that node, no need to update
+		return nil
+	}
+
+	reportedErrorNode := ncc.reportedErrorNode
+	if ncc.reportedErrorNode == "" && errorMsg != "" {
+		reportedErrorNode = nodeName
+	}
+	if ncc.reportedErrorNode == nodeName && errorMsg == "" {
+		// error for this node is fixed, report next error node
+		// zero value (aka empty string) will be returned if no more nodes have errors
+		reportedErrorNode, _ = sets.New(maps.Keys(ncc.nodeErrors)...).Delete(nodeName).PopAny()
```

Comment:
nit
```
for reportedErrorNode := range ncc.nodeErrors {
    if reportedErrorNode != nodeName {
        break
    }
}
```

Reply:
this will leave you with `reportedErrorNode = nodeName` if it is the only node in the errors map?
I can convert this to the loop though